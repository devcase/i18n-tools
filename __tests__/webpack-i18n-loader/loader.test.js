import compiler from "./__example__/compiler.js";

test.only("Test webpack-i18n", async () => {
  const stats = await compiler();
  const statsJson = stats.toJson();
  ['index.js', 'level1.js', 'level2.js', 'level4.js'].forEach(fileName => {
    const module = statsJson.modules.filter(x => x.id.indexOf(fileName) >= 0)[0]
    expect(module && module.source).toMatchSnapshot()
  })
});

test("Test webpack-i18n with optional chaining", async () => {
  const stats = await compiler({ entry: "./src/optionalChainingIndex.js" });
  const statsJson = stats.toJson();
  statsJson.modules.forEach(m => {
    expect(m.source).toMatchSnapshot();
  });
});
