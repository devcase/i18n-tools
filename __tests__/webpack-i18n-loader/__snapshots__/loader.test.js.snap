// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test webpack-i18n 1`] = `
"import Level1 from './level1?locale=en-US'
var text = \\"i18n:Meu texto internacionalizável\\"
console.log(Level1)"
`;

exports[`Test webpack-i18n 2`] = `
"//precisa ser atualizado para ./level2?locale=en-US
import sublevel from \\"./level2?locale=en-US\\";
export { default as level2 } from \\"./level2?locale=en-US\\"; //Traduzido para inglês

var text = \\"My i18n text\\";
export default sublevel;"
`;

exports[`Test webpack-i18n 3`] = `
"//precisa ser atualizado para ./level3?locale=en-US
import(\\"./level3?locale=en-US\\").then(t => console.log(t.default)); //Não pode mexer em ./level4 (excluído nas regras do webpack)

import sublevel from './level4'; //Traduzido para inglês

var text = \\"My i18n text\\";
export default sublevel;"
`;

exports[`Test webpack-i18n 4`] = `
"//Não modificado
var text = \\"i18n:Meu texto internacionalizável\\"

export default text
"
`;

exports[`Test webpack-i18n with optional chaining 1`] = `
"const f = () => {
  const o = \\"teste\\";
  console.log((o?.b?.c));
};

export default f;"
`;

exports[`Test webpack-i18n with optional chaining 2`] = `"import './optionalChaining?locale=en-US'"`;

exports[`Test webpack-i18n with optional chaining 3`] = `
"//precisa ser atualizado para ./level3?locale=en-US
import(\\"./level3\\").then(t => console.log(t.default))

//Não pode mexer em ./level4 (excluído nas regras do webpack)
import sublevel from './level4'

//Traduzido para inglês
var text = \\"i18n:Meu texto internacionalizável\\"

export default sublevel
"
`;

exports[`Test webpack-i18n with optional chaining 4`] = `
"//precisa ser atualizado para ./level3?locale=en-US
import(\\"./level3?locale=en-US\\").then(t => console.log(t.default)); //Não pode mexer em ./level4 (excluído nas regras do webpack)

import sublevel from './level4'; //Traduzido para inglês

var text = \\"My i18n text\\";
export default sublevel;"
`;

exports[`Test webpack-i18n with optional chaining 5`] = `
"//Traduzido para inglês
var text = \\"i18n:Meu texto internacionalizável\\"

export default text
"
`;

exports[`Test webpack-i18n with optional chaining 6`] = `
"//Traduzido para inglês
var text = \\"My i18n text\\";
export default text;"
`;

exports[`Test webpack-i18n with optional chaining 7`] = `
"//Não modificado
var text = \\"i18n:Meu texto internacionalizável\\"

export default text
"
`;

exports[`Test webpack-i18n with optional chaining 8`] = `
"const f = () => {
  const o = \\"teste\\";
  console.log((o?.b?.c));
};

export default f;"
`;

exports[`Test webpack-i18n with optional chaining 9`] = `"import './optionalChaining?locale=en-US'"`;
